oxigen_purchase/tests/test_oxigen_purchase.py

```diff
@@ -96,3 +96,87 @@ def test_02_orderpoint_considers_manual_po(self):
         self.assertEqual(po_line.product_qty, 20.0)
         in_progress = self.op1._quantity_in_progress().get(self.op1.id)
         self.assertEqual(in_progress, 20.0)
+
+    def test_01_picking_ref_in_po(self):
+
+        # Draft purchase order created
+        self.po = self.po_obj.create(
+            {
+                "partner_id": self.vendor_1.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.test_product.id,
+                            "product_qty": 8.0,
+                            "name": self.test_product.name,
+                            "product_uom": self.test_product.uom_id.id,
+                            "price_unit": 3.0,
+                            "date_planned": fields.Date.today(),
+                        },
+                    )
+                ],
+            }
+        )
+        self.assertTrue(self.po, "Purchase: no purchase order created")
+
+        # Purchase order confirm
+        self.po.button_confirm()
+
+        # Validate first shipment
+        self.picking = self.po.picking_ids[0]
+        self.picking.partner_ref = "ref 12"
+        self.assertEqual(
+            self.po.mapped("partner_ref"), ["ref 12"], "PO doesnt have a reference"
+        )
```

Comment:
you can extend the test changing now the reference in the picking again and checking the result is the expected.